File: 5.盛水最多的容器/5.py
# 前后指针
class Solution:
    def maxArea(self, height: List[int]) -> int:
        left = 0
        max_water = 0
        right = len(height)-1
        while(left<right):
            water = (right - left) *min(height[right],height[left])
            if(water>max_water):
                max_water = water
        
            if(height[right]>height[left]):
                left +=1
            else:
                right -= 1
        return max_water

File: 5.盛水最多的容器/5.cpp
#include <algorithm>
#include <iostream>
#include <vector>
using namespace std;
class Solution {
public:
    int maxArea(vector<int>& height) {
        int left = 0;
        int max_water = 0;
        int right = height.size() - 1;
        while(left<right)
        {
            int water = (right - left) * min(height[left],height[right]);
            if(water>max_water) max_water=water;
            if(height[left]>height[right]) right --;
            else left++;


        }
        return max_water;
    }
};